### DISCUSSION TITLE
Compile GDash with MSYS2 on WIndows

### DISCUSSION THREAD

**Discussion by herz-aus-gold-42** [revvv/gdash-export-CrLi#84] [Show and tell]
Here a HowTo for compile GDash with MSYS2 on Windows.
[Install-MYS2-GDash.txt](https://github.com/user-attachments/files/16574093/Install-MYS2-GDash.txt)

**Comment by revvv** [comment]
Looks good. Thank you so much!

**Comment by herz-aus-gold-42** [comment]
Hi, I try to configure and run make, without generate/modify files which are not in git.
So I create the build directory. (Why? -> the "build" directory is excluded in your ".gitignore" file)

$ cd ~
$ cd gdash-export-CrLi
$ mkdir build
$ cd build
$ .././configure

-> Makefile's and other stuff will be generated in the "build" directory, but src (and others) dir too 
-> So make don't find files
even when I define VPATH 
 export VPATH=~/gdash-export-CrL
 -> not work better
 
 Can you give me a hint how to compile without generate files which are not hidden with .gitignore

**Comment by revvv** [comment]
I don't know. I always build in root directory.
And I haven't added "build/*" in .gitignore. I just fetched the file from Zoltan's repo. Looks to me unnecessary. 

I guess you want to keep the sources directory clean. Just restore it after building.
I know there are some generated files checked in and some not. I did not tidy this up. Sometimes the generated files are helpful if your build fails.

**Reply by herz-aus-gold-42** [reply]
Ok, to be more precise:
After:
$ cd ~
$ cd gdash-export-CrLi
$ mkdir build
$ cd build
$ .././configure

All new files are generated in the "build" dir.
and repo is clean.
in "Makefile" the VPATH variable is set correctly:
VPATH = ./..
now I try "make" (in dir "build") 

The first bad thing is: make run configure again, and after that -> files modified in root and subdirs, mainly "Makefile.in"
Then Icon's can't be worked (VPATH is not considered):

`( rm -f icons.list; \
        for image in add_filled_rectangle.png add_freehand.png add_line.png add_join.png add_point.png add_raster.png add_rectangle.png add_fill_replace.png add_fill_border.png add_maze.png add_maze_uni.png add_maze_braid.png cave_editor.png move.png snapshot.png restart_level.png random_fill.png award.png to_top.png to_bottom.png object_on_all.png object_not_on_all.png object_not_on_current.png replay.png keyboard.png image.png statistics.png; do \
                echo $image | sed s/\.png$// >> icons.list; \
                echo $image >> icons.list; \
        done )
gdk-pixbuf-csource --rle --build-list `cat icons.list` > icons.cpp
failed to load "add_filled_rectangle.png": Failed to open file ΓÇ£add_filled_rectangle.pngΓÇ¥: No such file or directory
make[2]: *** [Makefile:761: icons.cpp] Error 1
make[2]: Leaving directory '/home/wiadm/gdash-export-CrLi/build/include'
make[1]: *** [Makefile:492: all-recursive] Error 1
make[1]: Leaving directory '/home/wiadm/gdash-export-CrLi/build'
make: *** [Makefile:391: all] Error 2
` 

Ok I understand solving this is a lot of work.
So I restore the files. Thanks.

**Comment by herz-aus-gold-42** [comment]
Here a modified ".gitignore" file. Upload as .txt file
After clone the repository, modify ".gitignore" to stay on a clean repo.
[gitignore.txt](https://github.com/user-attachments/files/16740265/gitignore.txt)
Copy file content to ".gitignore"

Then do "./configure" and "make" 
Restore the modified files (see comment's in gitignore.txt) - once
Now you can do "make", "make clean", "make distclean", "./configure" - as often as you like.
